kart/point_cloud/v1.py

```diff
@@ -40,3 +47,52 @@ def tilenames_with_lfs_paths(self):
         """Returns a generator that yields every tilename along with the path where the tile content is stored locally."""
         for blob_name, lfs_hash in self.tilenames_with_lfs_hashes():
             yield blob_name, get_local_path_from_lfs_hash(self.repo, lfs_hash)
+
+    def decode_path(self, path):
+        rel_path = self.ensure_rel_path(path)
+        if rel_path.startswith("tiles/"):
+            return ("tiles", self.decode_tile_path(rel_path))
+        return super().decode_path(rel_path)
+
+    def encode_tile_path(self, tilename, relative=False, *, schema=None):
+        """Given a tile's name, returns the path the tile's pointer should be written to."""
+        tile_prefix = hexhash(tilename)[0:2]
+        rel_path = f"tiles/{tile_prefix}/{tilename}"
+        return rel_path if relative else self.ensure_full_path(rel_path)
+
+    @classmethod
+    def decode_tile_path(cls, tile_path):
+        return tile_path.rsplit("/", maxsplit=1)[-1]
+
+    def diff(self, other, ds_filter=DatasetKeyFilter.MATCH_ALL, reverse=False):
+        """
+        Generates a Diff from self -> other.
+        If reverse is true, generates a diff from other -> self.
+        """
+        ds_diff = super().diff(other, ds_filter=ds_filter, reverse=reverse)
+        tiles_filter = ds_filter.get("tiles", ds_filter.child_type())
+        ds_diff["tiles"] = DeltaDiff(
+            self.diff_tiles(other, tiles_filter, reverse=reverse)
+        )
+        return ds_diff
+
+    def diff_tiles(self, other, tiles_filter=FeatureKeyFilter.MATCH_ALL, reverse=False):
+        """
+        Yields tiles deltas from self -> other, but only for tiles that match the tiles_filter.
+        If reverse is true, yields tiles deltas from other -> self.
+        """
+        yield from self.diff_subtree(
+            other,
+            "tiles",
+            key_filter=tiles_filter,
+            key_decoder_method="decode_tile_path",
+            value_decoder_method="get_tile_summary_from_pointer_blob",
+            reverse=reverse,
+        )
+
+    def get_tile_summary_from_pointer_blob(self, tile_pointer_blob):
+        # For now just return the blob contents as a string - it's a reasonable summary.
```

Comment:
This is certainly a good starting point.

Conceptually `pdal info --summary` (and `gdalinfo -json` for raster tiles) are probably more useful to see high level changes: tile has a different extent, etc

Do we have a concept of more-verbose diffs? (ala `--schema`,`--metadata`,`--statistics`,`--all` args to `pdal info` might be useful, but too verbose usually).